bluepyemodel/emodel_pipeline/plotting.py

```diff
@@ -1437,18 +1535,39 @@ def plot_FI_curves_comparison(
             color="grey",
             label="experiment",
         )
+        ax[1].plot(simulated_amp, simulated_freq, "o", color="blue", label="model")
         ax[1].set_xlabel("Amplitude (nA)")
         ax[1].set_ylabel("Voltage (mV)")
-        ax[1].set_title("IV curve (absolute amplitude)")
-        ax[1].plot(simulated_amp, simulated_freq, "o", color="blue", label="model")
-
-        ax[0].legend()
+        ax[1].set_title("FI curve (absolute amplitude)")
         ax[1].legend()
+
         if write_fig:
-            save_fig(
-                figures_dir,
-                emodel.emodel_metadata.as_string(emodel.seed) + "__FI_curve_comparison.pdf",
-            )
+            filename = f"{emodel.emodel_metadata.as_string(emodel.seed)}__FI_curve_comparison.pdf"
+            save_fig(figures_dir, filename)
+
+    updated_evaluator = copy.deepcopy(evaluator)
+    for emodel in emodels:
+        if not load_cells_for_emodel(emodel):
+            continue
+
+        expt_data = get_experimental_FI_curve_for_plotting(cells, prot_name, n_bin=n_bin)
+        expt_data_amp_rel = expt_data[0]
+        prot_name_original = get_original_protocol_name(prot_name, evaluator)
+        updated_evaluator = update_evaluator(
+            expt_data_amp_rel, prot_name_original, updated_evaluator
+        )
+
+    updated_evaluator.fitness_protocols["main_protocol"].execution_order = (
+        updated_evaluator.fitness_protocols["main_protocol"].compute_execution_order()
+    )
+
+    emodels = compute_responses(access_point, updated_evaluator, mapper, seeds)
+    for emodel in emodels:
+        expt_data = get_experimental_FI_curve_for_plotting(cells, prot_name, n_bin=n_bin)
```

Comment:
Is it ok not to use load_cells_for_emodel here in the loop before using cells variable?

Reply:
I had removed load_cells_for_emodel to reduce code duplication by using a global variable. However, I reverted back to the original logic since it is safer :)